infer/src/Makefile

```diff
@@ -122,15 +122,20 @@ doc: $(SRC_BUILD_COMMON) $(MAKEFILE_LIST)
 
 INFER_BIN_ALIASES = $(foreach alias,$(INFER_COMMANDS),$(BIN_DIR)/$(alias))
 
-$(INFER_BIN_ALIASES): Makefile
-#	make sure the bin directory exists
-	$(QUIET)$(MKDIR_P) $(@D)
+$(INFER_BIN_ALIASES): Makefile $(BIN_DIR)/$(INFER_MAIN)
+	$(REMOVE) $(@D)/$(@F)
```

Comment:
nit: Isn't `$(@D)/$(@F)` the same as `$@`?

Reply:
Indeed. I will simplify.